apm-agent-core/src/main/java/co/elastic/apm/agent/report/serialize/SerializationConstants.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.report.serialize;
+
+import co.elastic.apm.agent.configuration.CoreConfiguration;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.GlobalTracer;
+
+import javax.annotation.Nullable;
+
+public class SerializationConstants {
+
+    /**
+     * Matches default ZLIB buffer size.
+     * Lets us assume the ZLIB buffer is always empty,
+     * so that {@link DslJsonSerializer#getBufferSize()} is the total amount of buffered bytes.
+     */
+    public static final int BUFFER_SIZE = 16384;
+
+    public static final int MAX_VALUE_LENGTH = 1024;
+
+    @Nullable
+    private static volatile SerializationConstants INSTANCE;
+
+    private final int maxLongStringValueLength;
+
+    private SerializationConstants(int maxLongStringValueLength) {
+        this.maxLongStringValueLength = maxLongStringValueLength;
+    }
+
+    public static void init(CoreConfiguration coreConfiguration) {
+        INSTANCE = new SerializationConstants(coreConfiguration.getLongFieldMaxLength());
+    }
+
+    public static int getMaxLongStringValueLength() {
+        return get().maxLongStringValueLength;
+    }
+
+    private static SerializationConstants get() {
+        if (INSTANCE != null) {
+            return INSTANCE;
+        }
+
+        ElasticApmTracer tracer = GlobalTracer.getTracerImpl();
+        if (tracer == null) {
+            // should only happen before tracer is initialized, mostly in test
```

Comment:
[Minor] How many tests are affected by this?
I would prefer moving the initialization of `SerializationConstants` without a `GlobalTracer` being set to the tests themselves if it is only a handful of tests. Otherwise it "risks" that the `tracer==null` branch is accidentally executed even in production.

If it is a lot effort it's not worth it IMO, feel free to ignore this comment.

Reply:
This is a question for @SylvainJuge

Reply:
Only a few test classes actually require this, so I'll try pushing another version without this ugly lazy init.

Reply:
Should be better with 5d8b06e7fec0ce78ed5f6775b8b3c19bf9f5d4e0

Reply:
`MockReporter` required this initialization as well - [`c9bd607`](https://github.com/elastic/apm-agent-java/pull/3027/commits/c9bd6073edfafa6b9804a10110966b0fde9ca35e).
Not the prettiest, but I agree it's preferable over leaking this concern into production code, so we'll go with that and rethink if we find this biting us again.